fastapi/param_functions.py

```diff
@@ -238,8 +238,13 @@ def File(  # noqa: N802
     )
 
 
-def Depends(dependency: Callable = None) -> Any:  # noqa: N802
-    return params.Depends(dependency=dependency)
+DependencyType = TypeVar("DependencyType")
+
+
+def Depends(
+    dependency: Callable[..., DependencyType] = None
+) -> DependencyType:  # noqa: N802
+    return params.Depends(dependency=dependency)  # type: ignore
```

Comment:
This is the one downside of my suggest change. It makes the type signature wrong from the framework's point of view. However the upside is that it helps catch bugs in the consumer's code